<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>RFCTrainingDialog</class>
 <widget class="QDialog" name="RFCTrainingDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>731</width>
    <height>767</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Random Forest Classifier Training</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../qRandomForestClassifier.qrc">
    <normaloff>:/CC/plugin/qRandomForestClassifier/images/icon_train.png</normaloff>:/CC/plugin/qRandomForestClassifier/images/icon_train.png</iconset>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="dataGroupBox">
     <property name="title">
      <string>Data</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="5" column="2">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Class label</string>
        </property>
       </widget>
      </item>
      <item row="8" column="2">
       <widget class="QSpinBox" name="cloud2ClassSpinBox">
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="value">
         <number>2</number>
        </property>
       </widget>
      </item>
      <item row="13" column="0">
       <widget class="QLabel" name="label_12">
        <property name="text">
         <string>Class indices</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Role</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Cloud</string>
        </property>
       </widget>
      </item>
      <item row="13" column="1">
       <widget class="QLineEdit" name="classesLineEdit">
        <property name="toolTip">
         <string>Indices should be seperated by a space, e.g., &quot;0 1 2&quot;.</string>
        </property>
       </widget>
      </item>
      <item row="8" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>class #2</string>
        </property>
       </widget>
      </item>
      <item row="10" column="1">
       <widget class="QComboBox" name="cloudComboBox">
        <property name="toolTip">
         <string>The point cloud to be used for training.</string>
        </property>
       </widget>
      </item>
      <item row="13" column="2">
       <widget class="QSpinBox" name="numberClassesSpinBox">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>The number of classes detected (zero when the Class indices list is invalid).</string>
        </property>
        <property name="maximum">
         <number>1024</number>
        </property>
       </widget>
      </item>
      <item row="12" column="1">
       <widget class="QComboBox" name="scalarFieldComboBox">
        <property name="toolTip">
         <string>A scalar field of integer labels.</string>
        </property>
       </widget>
      </item>
      <item row="12" column="2">
       <widget class="QLabel" name="label_11">
        <property name="text">
         <string># of classes</string>
        </property>
       </widget>
      </item>
      <item row="7" column="2">
       <widget class="QSpinBox" name="cloud1ClassSpinBox">
        <property name="minimum">
         <number>1</number>
        </property>
       </widget>
      </item>
      <item row="8" column="1">
       <widget class="QComboBox" name="class2CloudComboBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>Points belonging to class #2</string>
        </property>
       </widget>
      </item>
      <item row="12" column="0">
       <widget class="QLabel" name="label_9">
        <property name="text">
         <string>Scalar field</string>
        </property>
       </widget>
      </item>
      <item row="7" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>class #1</string>
        </property>
       </widget>
      </item>
      <item row="10" column="0">
       <widget class="QLabel" name="label_10">
        <property name="text">
         <string>Cloud</string>
        </property>
       </widget>
      </item>
      <item row="7" column="1">
       <widget class="QComboBox" name="class1CloudComboBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>Points belonging to class #1</string>
        </property>
       </widget>
      </item>
      <item row="14" column="1" colspan="2">
       <widget class="QLabel" name="label_13">
        <property name="font">
         <font>
          <italic>true</italic>
         </font>
        </property>
        <property name="text">
         <string>Points labelled as -1 will be treated as unclassified.</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
        </property>
       </widget>
      </item>
      <item row="9" column="0" colspan="2">
       <widget class="QRadioButton" name="useScalarFieldsRadioButton">
        <property name="text">
         <string>Use scalar field as labels</string>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QRadioButton" name="usePointCloudsRadioButton">
        <property name="text">
         <string>Use point clouds</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="scalesGroupBox">
     <property name="title">
      <string>Scales</string>
     </property>
     <layout class="QFormLayout" name="formLayout_2">
      <item row="0" column="1">
       <widget class="QFrame" name="scalesRampFrame">
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QSpinBox" name="numberScalesSpinBox">
           <property name="toolTip">
            <string>The number of scales to compute.</string>
           </property>
           <property name="prefix">
            <string>n = </string>
           </property>
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>20</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="minScaleDoubleSpinBox">
           <property name="toolTip">
            <string>The smallest scale used as a voxel size for the planimetric grid (if the default value -1 is used, its value is automatically estimated).</string>
           </property>
           <property name="prefix">
            <string>min scale = </string>
           </property>
           <property name="decimals">
            <number>6</number>
           </property>
           <property name="minimum">
            <double>0.000000000000000</double>
           </property>
           <property name="maximum">
            <double>1000000000.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.100000000000000</double>
           </property>
           <property name="value">
            <double>0.000000000000000</double>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Features</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="1">
       <widget class="QPushButton" name="addPushButton">
        <property name="text">
         <string>Add</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0" rowspan="3">
       <widget class="QListView" name="featureListView"/>
      </item>
      <item row="1" column="1">
       <widget class="QPushButton" name="removePushButton">
        <property name="text">
         <string>Remove</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Advanced</string>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <item row="1" column="0">
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>Max core points</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QSpinBox" name="maxPointsSpinBox">
        <property name="toolTip">
         <string>Maximum number of core points computed on each class</string>
        </property>
        <property name="minimum">
         <number>0</number>
        </property>
        <property name="maximum">
         <number>1000000000</number>
        </property>
        <property name="singleStep">
         <number>1000</number>
        </property>
        <property name="value">
         <number>100000</number>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QLabel" name="label_8">
        <property name="font">
         <font>
          <italic>true</italic>
         </font>
        </property>
        <property name="text">
         <string>When set to 0 all points will be used to train the classifier.</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_14">
        <property name="text">
         <string># of trees</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QSpinBox" name="nTreesSpinBox">
        <property name="toolTip">
         <string>The number of trees generated by the training algorithm. Higher values may improve result at the cost of higher computation times (in general, using a few dozens of trees is enough).</string>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>1024</number>
        </property>
        <property name="value">
         <number>25</number>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="label_15">
        <property name="text">
         <string>Maximum tree depth</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QSpinBox" name="maxTreeDepthSpinBox">
        <property name="toolTip">
         <string>The maximum depth of the trees. Higher values will improve how the forest fits the training set. A overly low value will underfit the test data and conversely an overly high value will likely overfit.</string>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>1024</number>
        </property>
        <property name="value">
         <number>20</number>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QCheckBox" name="evaluateParamsCheckBox">
        <property name="toolTip">
         <string>Check this to add more points to the 2D classifier behavior representation</string>
        </property>
        <property name="text">
         <string>Show classifier behavior on </string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QComboBox" name="evaluationCloudComboBox">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="toolTip">
         <string>A point cloud that may be used after training the classifier to quickly see the results (equivalent to calling the Train, then Classify functions individually).</string>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="QFrame" name="frame">
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_8">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>1</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>class1CloudComboBox</tabstop>
  <tabstop>cloud1ClassSpinBox</tabstop>
  <tabstop>class2CloudComboBox</tabstop>
  <tabstop>cloud2ClassSpinBox</tabstop>
  <tabstop>minScaleDoubleSpinBox</tabstop>
  <tabstop>maxPointsSpinBox</tabstop>
  <tabstop>evaluateParamsCheckBox</tabstop>
  <tabstop>evaluationCloudComboBox</tabstop>
  <tabstop>buttonBox</tabstop>
 </tabstops>
 <resources>
  <include location="../qRandomForestClassifier.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>RFCTrainingDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>RFCTrainingDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>evaluateParamsCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>evaluationCloudComboBox</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>109</x>
     <y>281</y>
    </hint>
    <hint type="destinationlabel">
     <x>343</x>
     <y>282</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
